### ISSUE TITLE
$name is detected as a top-scope variable

### ISSUE THREAD

**Issue by hkraal** [rodjek/puppet-lint#23]
The snippet:

``` ruby
define foo::bar( $var ) {
  exec { "echo-test-${name}":
    unless "grep ${var} /tmp/foo_bar.txt"
    command "echo ${name} >> /tmp/foo_bar.txt"
  }
}
```

will give the errors:

```
WARNING: top-scope variable being used without an explicit namespace on line 2
WARNING: top-scope variable being used without an explicit namespace on line 4
```

which shouldn't as far as I understand. Probably a bug to the exceptional nature of the $name variable?
